### PR TITLE
wifi-2940: Add ath11k beacon and mcast rate

### PR DESCRIPTION
**Pull request by chaitanyagodavarthi**
Add support for changing mcast and beacon rate
in opensync for ath11k
Add debugfs entry for ath11k driver to set mcast,
beacon and mgmt rate.
For help:
cat /sys/kernel/debug/ieee80211/phyX/ath11k/set_rates

Eg: Set beacon rate to 12Mbps for iface wlan0 5G band
echo "wlan0 beacon 5 0x10000002" > /sys/kernel/debug/ieee80211/phy0/ath11k/set_rates

Signed-off-by: Chaitanya Godavarthi <chaitanya.kiran@netexperience.com>

### CODE REVIEW

File: feeds/wifi-ax/mac80211/patches/pending/214-debugfs-set-rate-mgmt-mcast-bcn.patch [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
++	}
++	else if (set_rate_type == WMI_VDEV_PARAM_BCAST_DATA_RATE) {
++		arvif->bcast_rate[cfg_band] = rc;
++		arvif->bcast_rate_set[cfg_band] = true;
++	}
++	else if (set_rate_type == WMI_VDEV_PARAM_MCAST_DATA_RATE) {
++		arvif->mcast_rate[cfg_band] = rc;
++		arvif->mcast_rate_set[cfg_band] = true;
++	}
++	else if (set_rate_type == WMI_VDEV_PARAM_BEACON_RATE) {
```
Comment (ArifAlam) [inline feeds/wifi-ax/mac80211/patches/pending/214-debugfs-set-rate-mgmt-mcast-bcn.patch:175]: duplicate entry.

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c [hunk 2]
```diff
@@ -200,14 +260,9 @@ int rrm_get_backup_channel(const char * freq_band)
 
 }
 
-bool rrm_noreload_config(struct schema_Wifi_RRM_Config *conf, const char *rname)
+bool rrm_noreload_config(struct schema_Wifi_RRM_Config *conf, const char *wlanif)
```
Comment (shimmy568) [inline feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c:263]: Not really part of this PR. But I think we can change this function to void, since we aren't actually returning anything else other than true 

(it looks like that if statement that this was called in before didn't even have anything in it in the first place)

**Review by shimmy568 — APPROVED** [review]

File: feeds/wifi-ax/mac80211/patches/pending/214-debugfs-set-rate-mgmt-mcast-bcn.patch (hunk 1 above)
Comment (chaitanyagodavarthi) [reply feeds/wifi-ax/mac80211/patches/pending/214-debugfs-set-rate-mgmt-mcast-bcn.patch:175]: done

**Review by yashvardhan-netex — COMMENTED** [review]
Tested on wifi5? Because there are some changes in the hotplug script.

**Comment by chaitanyagodavarthi** [comment]
> Tested on wifi5? Because there are some changes in the hotplug script.

Yes, Tested on wifi5

**Review by yashvardhan-netex — APPROVED** [review]
Looks good!

File: feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c (hunk 2 above)
Comment (chaitanyagodavarthi) [reply feeds/wlan-ap/opensync/src/platform/openwrt/src/lib/target/src/rrm_config.c]: Done

**Comment by ArifAlam** [comment]
Merged to pending. Thanks.
